src/topology/locally_constant/basic.lean

```diff
@@ -426,4 +426,55 @@ lemma coe_desc {X α β : Type*} [topological_space X] (f : X → α) (g : α 
 
 end desc
 
+section char_fn
+variables (R : Type*) [mul_zero_one_class R]
+
+/-- Characteristic functions are locally constant functions taking `x : X` to `1` if `x ∈ U`,
+  where `U` is a clopen set, and `0` otherwise. -/
+noncomputable def char_fn {U : set X} (hU : is_clopen U) : locally_constant X R :=
```

Comment:
Can you connect this to https://leanprover-community.github.io/mathlib_docs/algebra/indicator_function.html#set.indicator ?
I imagine `to_fun` could be `set.indicator U 1`. If you tag it with `@[simps]` then there will automatically be simp-lemmas that rewrite the coercion-to-function of this definition to `set.indicator`.

Here's another idea/suggestion: Define `locally_constant.indicator`, which takes a `locally_constant` function `f` and a clopen set `U`, and returns the locally constant function `f` on `U`, and `0` outside of `U`.
Then the characteristic function is the special case where `f` is the constant function `1`.

Reply:
Thank you for the suggestions! I have used the second one, please let me know if it looks okay!